src/ops/convolution.rs

```diff
@@ -0,0 +1,495 @@
+use crate::error::OperationError;
+use crate::{Array, StreamOrDevice};
+use mlx_macros::default_device;
+
+#[inline]
+fn conv_general_device_inner(
+    array: &Array,
+    weight: &Array,
+    strides: &[i32],
+    padding_lo: &[i32],
+    padding_hi: &[i32],
+    kernel_dilation: &[i32],
+    input_dilation: &[i32],
+    groups: i32,
+    flip: bool,
+    stream: StreamOrDevice,
+) -> Array {
+    unsafe {
+        Array::from_ptr(mlx_sys::mlx_conv_general(
+            array.as_ptr(),
+            weight.as_ptr(),
+            strides.as_ptr(),
+            strides.len(),
+            padding_lo.as_ptr(),
+            padding_lo.len(),
+            padding_hi.as_ptr(),
+            padding_hi.len(),
+            kernel_dilation.as_ptr(),
+            kernel_dilation.len(),
+            input_dilation.as_ptr(),
+            input_dilation.len(),
+            groups,
+            flip,
+            stream.as_ptr(),
+        ))
+    }
+}
+
+/// General convolution over an input with several channels.
+///
+/// > Only 1d and 2d convolutions are supported at the moment
```

Comment:
This put a tab on the generate doc, which is not common in the rust community. I'd prefer to avoid it if possible, but it's up to you.